test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,295 @@
+     * @throws IOException
+     * @throws ClassNotFoundException
      */
-    private static void toFile(Serializable o, String filePath) throws IOException {
-        FileOutputStream fout = new FileOutputStream(filePath, true);
-        ObjectOutputStream oos = new ObjectOutputStream(fout);
-        oos.writeObject(o);
-        oos.close();
+    @Test(dataProvider="GregorianCalendarDataBase64")
+    public void testReadCalBase64(String javaVersion, GregorianCalendarAndDurationSerData gcsd, String gregorianDate) throws IOException,
+            ClassNotFoundException {
+        final ByteArrayInputStream bais = new ByteArrayInputStream(Base64.getDecoder().decode(gcsd.getGregorianCalendarBase64()));
+        final ObjectInputStream ois = new ObjectInputStream(bais);
+        final XMLGregorianCalendar xgc = (XMLGregorianCalendar) ois.readObject();
+        Assert.assertEquals(xgc.toString(), gregorianDate);
     }
 
-    static String errMessage;
-    int passed = 0, failed = 0;
+    /**
+     * verify serialization compatibility for Duration
+     * @param javaVersion
+     * @param gcsd
+     * @param duration
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    @Test(dataProvider = "DurationData")
+    public void testReadDurationBytes(String javaVersion, GregorianCalendarAndDurationSerData gcsd, String duration) throws IOException,
+            ClassNotFoundException {
+        final ByteArrayInputStream bais = new ByteArrayInputStream(gcsd.getDurationBytes());
+        final ObjectInputStream ois = new ObjectInputStream(bais);
+        final Duration d1 = (Duration) ois.readObject();
+        Assert.assertEquals(d1.toString().toUpperCase(), duration);
+    }
 
-    void fail(String errMsg) {
-        if (errMessage == null) {
-            errMessage = errMsg;
-        } else {
-            errMessage = errMessage + "\n" + errMsg;
-        }
-        failed++;
+    /**
+     * verify serialization compatibility for Duration
+     * @param javaVersion
+     * @param gcsd
+     * @param duration
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    @Test(dataProvider = "DurationDataBase64")
+    public void testReadDurationBase64(String javaVersion, GregorianCalendarAndDurationSerData gcsd, String duration) throws IOException,
+            ClassNotFoundException {
+        final ByteArrayInputStream bais = new ByteArrayInputStream(Base64.getDecoder().decode(gcsd.getDurationBase64()));
+        final ObjectInputStream ois = new ObjectInputStream(bais);
+        final Duration d1 = (Duration) ois.readObject();
+        Assert.assertEquals(d1.toString().toUpperCase(), duration);
     }
 
-    void success(String msg) {
-        passed++;
-        System.out.println(msg);
+    /**
+     * Generates the Java Pseudo code for serialized Gregorian Calendar byte array that can be cut & pasted into the
+     * JDK<version>GregorianCalendarAndDurationSerData.java files.
+     * @param baos
+     */
```

Comment:
I think there needs to be a general comment describing how these methods are used to create the JDK<version>GregorianCalendarAndDurationSerData.java files.  There should also be a description/comment for the methods defined in GregorianCalendarAndDurationSerData.java

We need to try and put ourselves in the place of a future maintainer who needs to understand how to  create a version of one of these files.

You could probably also create a method which generates a JDK<version>GregorianCalendarAndDurationSerData.java file to save the developer from multiple cut an pastes.

At a minimum, there really should be a step by step guide

Reply:
How these methods can be used, added comment regarding this under setup method. Added description for the methods defined in GregorianCalendarAndDurationSerData.java file. Now developer should be able to create the <JDK version >GregorianCalendarAndDurationSerData.java file after reading these comments.

Reply:
Please see comment above in the setUp method as I feel we can do more here